src/parser/permit-generation-module.ts

```diff
@@ -0,0 +1,120 @@
+import { context } from "@actions/github";
+import { Value } from "@sinclair/typebox/value";
+import { createClient } from "@supabase/supabase-js";
+import {
+  Context,
+  createAdapters,
+  Database,
+  encodePermits,
+  generatePayoutPermit,
+  SupportedEvents,
+  TokenType,
+} from "@ubiquibot/permit-generation";
+import configuration from "../configuration/config-reader";
+import contentEvaluatorConfig from "../configuration/content-evaluator-config";
+import { getOctokitInstance } from "../get-authentication-token";
+import { IssueActivity } from "../issue-activity";
+import { Module, Result } from "./processor";
+import program from "./command-line";
+
+interface Payload {
+  evmNetworkId: number;
+  issueUrl: string;
+  evmPrivateEncrypted: string;
+  issue: { id: number };
+}
+
+export class PermitGenerationModule implements Module {
+  readonly _configuration: PermitGenerationModule = configuration.permitGeneration;
+  readonly _supabase = createClient<Database>(process.env.SUPABASE_URL, process.env.SUPABASE_ANON_KEY);
+
+  async transform(data: Readonly<IssueActivity>, result: Result): Promise<Result> {
+    const payload: Context["payload"] & Payload = {
+      ...context.payload.inputs,
+      issueUrl: program.opts().issue,
+      evmPrivateEncrypted: program.opts().evmPrivateEncrypted,
+      evmNetworkId: program.opts().evmNetworkId,
+    };
+    const issueId = Number(payload.issueUrl.match(/[0-9]+$/)?.[1]);
+    payload.issue = {
+      id: issueId,
+    };
+    const env = process.env;
+    const eventName = context.eventName as SupportedEvents;
+    const octokit = getOctokitInstance();
+    const logger = {
+      debug(message: unknown, optionalParams: unknown) {
+        console.log(message, optionalParams);
+      },
+      error(message: unknown, optionalParams: unknown) {
+        console.error(message, optionalParams);
+      },
+      fatal(message: unknown, optionalParams: unknown) {
+        console.error(message, optionalParams);
+      },
+      info(message: unknown, optionalParams: unknown) {
+        console.log(message, optionalParams);
+      },
+      warn(message: unknown, optionalParams: unknown) {
+        console.warn(message, optionalParams);
+      },
+    };
+    const adapters = {} as ReturnType<typeof createAdapters>;
+
+    for (const [key, value] of Object.entries(result)) {
+      try {
+        const config: Context["config"] = {
+          evmNetworkId: payload.evmNetworkId,
+          evmPrivateEncrypted: payload.evmPrivateEncrypted,
+          permitRequests: [
+            {
+              amount: value.total,
+              userId: value.userId,
+              contributionType: "",
```

Comment:
It's a good question. I had changed the system to identity the contribution level of the users. I think it would be better if I pass down the new value there either way, I will change that.

Reply:
it's used in NFT

Reply:
@whilefoo I meant that we do not even give the URL to claim ERC721 so these permits are not claimable by users (afaik).
From what I understand from the previous codebase, it holds the kind of contributions the user is accountable for
<img width="287" alt="image" src="https://github.com/ubiquibot/conversation-rewards/assets/9807008/11f7ddec-e3fd-4a21-8604-2dab37a81cc6">
I'll put a similar value with the new codebase.